### PR TITLE
Refactor `WormholeProvider`

### PR DESCRIPTION
**Pull request by JustusFT**

### CODE REVIEW

**Comment by donpui** [comment]
If we are getting rid of worker, we need to cleanup in gulpfile.js and Dockerfile as there are some tasks defined.

**Comment by donpui** [comment]
@JustusFT I have issue, that I cannot download file, when I use url on receiver side: http://localhost:8080/#6-backwater-southward. 
If I enter code manually, download works.

File: client/src/app/components/providers/WormholeProvider.tsx [hunk 1]
```diff
@@ -1,225 +1,312 @@
...
+      file: FileInfo;
+      progress: number;
+      // TODO: only exist on failed step
+      error?: any;
+      // TODO: make step before "inProgress"?
+      startTime?: number;
+    }
+  | {
+      status: "receiving";
+      step: "confirming" | "inProgress" | "failed" | "succeeded";
```
Comment (donpui) [inline client/src/app/components/providers/WormholeProvider.tsx:33]: Should we have rejection/cancellation step?

**Comment by donpui** [comment]
One more error, not handled case when  `Nameplate is unclaimed: 14`.

**Review by donpui — CHANGES_REQUESTED** [review]
I have pushed small gulpfile changes, however we need to address other issues too.

Overall simplification is good here. However I am think, that in future we might (or not) get back to worker, when dilation might appear, especially if we want several files sending simultaneously or keep them in background and allowing navigate in app. But may be it is easier to start with worker from fresh with specific features which we want to achieve, then what we have now.

**Comment by donpui** [comment]
After discussion with @JustusFT we agreed to close this PR, as it requires more dev and testing effort to finalise, however with dilation, we might still need to get back to worker. Closing PR, but keeping branch for future work.

**Comment by meejah** [comment]
Is there ticket describing what this PR was / is trying to achieve (so that when/if we do revisit it, some context can be obtained without reading all ~2000 LoC of the PR)?

**Pull request closed without merging**